content/lua/concepts/strings/terms/find/find.md

```diff
@@ -0,0 +1,176 @@
+---
+Title: '.find()'
+Description: 'The **`.find()`** function is used to search for a substring within a string and returns the start and end indices of the first occurrence of the substring.'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+  - 'Game Development'
+Tags:
+  - 'Functions'
+  - 'Index'
+  - 'Methods'
+  - 'Search'
+  - 'Strings'
+CatalogContent:
+  - 'learn-lua'
+  - 'paths/computer-science'
+---
+
+The **`.find()`** function is used to search for a substring within a string and returns the start and end indices of the first occurrence of the substring.
+
+If the substring is not found, the function returns `nil`.
+
+## Syntax
+
+These are 2 different syntaxes for the .find() function:
+
+### 1
+
+```pseudo
+string.find(fullString, searchString, init, pattern)
+```
+
+This requires assigning the function to a variable, to return the function.
+
+or
+
+### 2
+
+```pseudo
+startindex, endindex = string.find(fullstring, searchstring, init, pattern)
+```
+
+This uses multiple assignment to save the results to an individual variables.
+
+- `fullString`: Is the string we are searching through.
+- `searchString`: Is the sub-string to search for.
+- `init` (optional): Is the initial index where the search starts. *Default index is 1*.
+- `pattern` (optional): Is a Boolean flag indicating whether to search using **pattern matching**[^1]. When set to false, pattern matching can be used, where as when set to true, pattern matching is ignored, and exact matches are searched for. *Default is false*.
+
+## Example 1
+
+Use `.find()` without multiple assignments:
+
+```lua
+sentence = 'hello world'
+search1 = string.find(sentence, "hello")
+search2 = string.find(sentence, "world")
+search3 = string.find(sentence, "universe")
+
+print(search1)
+print(search2)
+print(search3)
+```
+
+The above example assigns the function to a unique variable, in order for the results to be returned/printed to the console.
+
+The results of the above functions are:
+
+```shell
+1
+7
+nil
+```
+
```

Comment:
```suggestion

Explaining how each output came:
```

Reply:
While I can see why this could be a way of introducing the explanations, I think that having the explanations directly after the code example reduces the need for an extra line, when context clues can indicate that the following explanation is referring to the previous code.

I have found some instances of no intro within the live docs [here](https://www.codecademy.com/resources/docs/java/strings/intern), [here](https://www.codecademy.com/resources/docs/python/strings/find) and [here](https://www.codecademy.com/resources/docs/cpp/strings/find). 

Is it possible to not need the line?

Reply:
Yes it will work

Reply:
thank you